static/scripts/rewards/helpers.ts

```diff
@@ -36,14 +35,12 @@ const RPC_HEADER = {
   "Content-Type": "application/json",
 };
 
-export const getErc20Contract = async (contractAddress: string, networkId: number): Promise<Contract> => {
-  const providerUrl = await getOptimalRPC(networkId);
-  const provider = new ethers.providers.JsonRpcProvider(providerUrl);
+export const getErc20Contract = async (contractAddress: string, provider: JsonRpcProvider): Promise<Contract> => {
   const contractInstance = new ethers.Contract(contractAddress, erc20Abi, provider);
   return contractInstance;
 };
 
-export const getOptimalRPC = async (networkId: number): Promise<string> => {
+export const getOptimalProvider = async (networkId: number) => {
```

Comment:
Hey there @Keyrxng I plan to merge this but please run `yarn format` and pass the lint errors first!

Reply:
prettier was failing for me trying to parse the /lib folder which I've excluded from both prettier and cspell as it's not required/creates a lot of noise esp cspell